.github/workflows/publish-HOWTOs.yml

```diff
@@ -0,0 +1,86 @@
+name: Publish HOWTOs
+on:
+  - push
+  - workflow_dispatch
+jobs:
+  build:
+    name: Build and prepare HOWTOs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14
+      - name: Install dependencies
+        run: npm install
+      - name: Build packages
+        run: npm run build
+      - name: Prepare for publishing (GitHub Pages)
+        run: npm run prep
+        env:
+          PKG_HOWTOS_LOCATION: github
+          PKG_HOWTOS_PATH: ${{ github.ref_name }}
+      - name: Prepare for publishing (AWS s3 CDN)
+        run: npm run prep
+        env:
+          PKG_HOWTOS_LOCATION: aws
+          PKG_HOWTOS_PATH: ${{ github.ref_name }}
+      - name: Upload artifacts (GitHub Pages)
+        uses: actions/upload-artifact@v3
+        with:
+          if-no-files-found: error
+          name: workflow-howtos-github
+          path: public-github/
+          retention-days: 1
+      - name: Upload artifacts (AWS s3 CDN)
+        uses: actions/upload-artifact@v3
+        with:
+          if-no-files-found: error
+          name: workflow-howtos-aws
+          path: public-aws/
+          retention-days: 1
+
+  publish-gh:
+    name: Publish HOWTOs (GitHub Pages)
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: workflow-howtos-github
+          path: public-github/
+      - name: Publish
+        uses: JamesIves/github-pages-deploy-action@4.1.7
+        with:
+          branch: gh-pages # The branch the action should deploy to.
+          folder: public-github # The folder the action should deploy
+          target-folder: ${{ github.ref_name }}
+
+
+  publish-aws:
```

Comment:
Is there a way of only publishing specific branches to aws? e.g. we would want all of the workspace versioned branches but not the other branches (we can publish to github for testing and sharing but we wouldn't want to clutter the cdn with).

Reply:
Sure. I will add a selector that matches `^workspace/[0-9]+\.[0-9]+\.[0-9]+$`.

Reply:
Ok. So apparently regexp matching is not a thing in step conditional expressions. At least not out-of-the-box. I could do it with a *helper step* that does the regexp match and outputs a *boolean* to check against. Or I could use the existing conditional expression functions and match against a substring, e.g. `if: startsWith(github.ref_name, "workspace/")` - that is assuming that `workspace/` prefix is reserved for production deployments. Please advise how you would prefer to proceed.

Reply:
Changed in https://github.com/built-on-openfin/workspace-starter/pull/105/commits/37b25882b1cb2a0ee04af477bc40d29d4c6932a0 (using the *prefix* scheme). If you need it to be a full regexp please let me know.

Reply:
By the way, would you like me to clean up the current s3 content in the `samples.openfin.co` bucket?

Reply:
Hi yes, the prefix should be fine as only the workspace/ versions are what we would encourage people to use.

If there is stuff on samples.openfin.co please feel free to clean it up thanks!